templates/index.html

```diff
@@ -235,8 +251,25 @@ <h1>E-bike routing Metro Manila</h1>
                 polyLineGroup.clearLayers();
             }
 
+
+            if (sourceCircleMarker) map.removeLayer(sourceCircleMarker);
+            if (destMarker) map.removeLayer(destMarker);
+
             // Only run the route when both Source and Target coordinates are present
             if (sourceCoordinates && targetCoordinates) {
+
+                sourceCircleMarker = L.circleMarker([sourceCoordinates.lat, sourceCoordinates.lng], { color: 'white' }).addTo(map);
+                destMarker = L.marker([targetCoordinates.lat, targetCoordinates.lng], {
+                    icon: L.icon({
+                        iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
+                        shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+                        iconSize: [25, 41],
+                        iconAnchor: [12, 41],
+                        popupAnchor: [1, -34],
+                        shadowSize: [41, 41]
+                    })
+                }).addTo(map);
```

Comment:
Markers are only added when mapRoute is called, but this only executes when sourceCoordinates and targetCoordinates exists